tests.py

```diff
@@ -808,19 +809,31 @@ def test_chain_del_2(self):
     def test_expire_context(self):
         """Test adding keys with an expire value by using the contextmanager."""
         with self.r.expire_at(3600):
-            self.r['foobar'] = 'barbar'
+            self.r['foobar1'] = 'barbar1'
+        print(type(timedelta(hours=0.25, minutes=15, seconds=900)))
+        with self.r.expire_at(timedelta(hours=0.25, minutes=15, seconds=900)):
+            self.r['foobar2'] = 'barbar2'
 
-        actual_ttl = self.redisdb.ttl('{}:foobar'.format(TEST_NAMESPACE_PREFIX))
+        actual_ttl = self.redisdb.ttl('{}:foobar1'.format(TEST_NAMESPACE_PREFIX))
         self.assertAlmostEqual(3600, actual_ttl, delta=2)
+        actual_ttl = self.redisdb.ttl('{}:foobar2'.format(TEST_NAMESPACE_PREFIX))
+        self.assertAlmostEqual(2700, actual_ttl, delta=2)
```

Comment:
Great job! However, let's distinguish between the current test and the new functionality tests. The reason for doing so is as follows:
Each test plays a crucial role in ensuring the stability of the project. By creating separate tests for every new functionality, we will have the flexibility to add and remove features in the future without having to modify our unit tests.
This approach also makes each Pull Request more self-contained.

Could you consider the code below?
```
  def test_expire_context_timedelta(self):
         """
         Test adding keys with an expire value by using the contextmanager.
         With timedelta as argument.
         """
         minute_in_seconds = 60
         hour_in_seconds = 60 * 60
         
         one_hour_timedelta = timedelta(hours=1)
         with self.r.expire_at(one_hour_timedelta):
             self.r['one_hour'] = 'one_hour'
             
         one_minute = timedelta(minutes=1)
         with self.r.expire_at(one_hour_timedelta):
             self.r['one_minute'] = 'one_minute'

         actual_ttl = self.redisdb.ttl('{}:one_hour'.format(TEST_NAMESPACE_PREFIX))
         self.assertAlmostEqual(hour_in_seconds, actual_ttl, delta=2)
         
         actual_ttl = self.redisdb.ttl('{}:one_minute'.format(TEST_NAMESPACE_PREFIX))
         self.assertAlmostEqual(minute_in_seconds,  actual_ttl, delta=2)
  ```